src/vec_int64_ppc.h

```diff
@@ -291,7 +291,7 @@ vec_cmpequd (vui64_t a, vui64_t b)
        rr = vec_perm (r, r, permute);
        r= vec_and (r, rr);
     }
-  result = (vui64_t)r;
+  result = (vb64_t)r;
```

Comment:
In a few other places, you removed a cast like this. Do you need it here?

Reply:
No, in this case r and the vec_sel inputs are __vector unsigned int to work correctly on POWER7.